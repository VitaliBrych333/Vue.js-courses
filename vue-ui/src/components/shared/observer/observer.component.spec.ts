import { mount } from '@vue/test-utils'
import Observer from './observer.component.vue'

const intersectionObserverMock = () => ({
  observe: () => null,
  disconnect: () => null
})
window.IntersectionObserver = jest
  .fn()
  .mockImplementation(intersectionObserverMock)

describe('Observer.vue', () => {
  it('should render correctly', () => {
    const wrapper = mount(Observer)

    expect(wrapper.text()).toMatchSnapshot()
  })

  it('should call destroyed', () => {
    const stub = jest.fn()
    const wrapper = mount(Observer, {
      destroyed() {
        stub()
      }
    }).destroy()

    expect(stub).toBeCalled()
  })
})
