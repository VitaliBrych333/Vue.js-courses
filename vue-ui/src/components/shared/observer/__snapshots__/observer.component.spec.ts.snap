// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Observer.vue should render correctly 1`] = `""`;
